contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,197 @@
+        URI rootURI = new URI(target.replaceAll("(?<=:[0-9]{1,5})/.*", ""));
+        return FileSystem.get(rootURI, new JobConf());
+    }
+
+    private void deleteDir(FileSystem fs, String target) throws Exception {
+        // Extract directory path from URL, e.g. hdfs://hostname:9000/a/b/c => /a/b/c
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        if (fs.exists(path)) {
+            fs.delete(path, true);
+            if (fs.exists(path)) {
+                warn("Could not delete temp directory " + path + " in CDN!");
+            } else {
+                info("Deleted " + path);
+            }
+        }
+    }
+
+    private void deleteDirOnExit(FileSystem fs, String target) throws Exception {
+        // Extract directory path from URL, e.g. hdfs://hostname:9000/a/b/c => /a/b/c
+        Path path = new Path(target.replaceAll(".*://.*?(?=/)", ""));
+
+        fs.deleteOnExit(path);  // Delete the directory even if an exception occurs
+        info(path + " is scheduled to be deleted on exit.");
+    }
+
+    private void addPermissionsToParents(FileSystem fs, String target) throws Exception {
+        // Extract directory path from URL, e.g. hdfs://hostname:9000/a/b/c => a/b/c
+        String pathFromRoot = target.replaceAll(".*://.*?/", "");
+        String parent = "";
+
+        for (String seg: pathFromRoot.split("/")) {
+            parent = parent + "/" + seg;
+            Path parentPath = new Path(parent);
+            FsPermission perm = fs.getFileStatus(parentPath).getPermission();
+            FsAction u = perm.getUserAction();
+            FsAction g = perm.getGroupAction();
+            FsAction o = perm.getOtherAction();
+            boolean changed = false;
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READBYGROUP, false)) {
+                if (!g.implies(FsAction.READ_EXECUTE)) {
+                    g = g.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READBYOTHER, false)) {
+                if (!o.implies(FsAction.READ_EXECUTE)) {
+                    o = o.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (changed) {
+                FsPermission desiredPerm = new FsPermission(u, g, o);
+                fs.setPermission(parentPath, desiredPerm);
+                assert fs.getFileStatus(parentPath).getPermission().equals(desiredPerm);
+                info("for path " + parent + ", permissions changed from: " + perm + " to: " + desiredPerm);
+            }
+        }
+    }
+
+    private String pickCDN() {
+        List<String> pushClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CLUSTER);
+        List<String> cdnClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER);
```

Comment:
I like how you're prefixing all of these config constants by `PUSH_CDN`. That makes the organization easier to grasp. Let's make sure we really do have this prefix everywhere, for all CDN-related configs.